packages/types/src/discord/channel.ts

```diff
@@ -1,12 +1,13 @@
 /** Types for: https://discord.com/developers/docs/resources/channel */
 
 import type { DiscordMessageComponents } from './components.js'
+import type { DiscordThreadCreateExtra } from './gateway.js'
 import type { DiscordMember } from './guild.js'
 import type { DiscordAllowedMentions, DiscordAttachment, DiscordEmbed, MessageFlags } from './message.js'
 import type { DiscordUser } from './user.js'
 
 /** https://discord.com/developers/docs/resources/channel#channel-object-channel-structure */
-export interface DiscordChannel {
+export interface DiscordChannel extends Partial<DiscordThreadCreateExtra> {
```

Comment:
Does this needs to be Partial when `DiscordThreadCreateExtra`'s `newly_created?: boolean` is already key optional?

```suggestion
export interface DiscordChannel extends DiscordThreadCreateExtra {
```

Reply:
I'd rather keep using Partial even if it isn't doing anything,  if discord were to add something that always exists to this Extra we would need to add the partial back
(and anyway these extends are probably going away in another pr, we (I and stickz seemed to agree) want to split the gateway event type from the type that may come from the rest api)